tracer/test/test-applications/debugger/Directory.Build.props

```diff
@@ -1,19 +1,19 @@
 <Project>
   <PropertyGroup>
-    <!-- only run .NET Framework tests on Windows -->
-    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">net461;netcoreapp2.1;netcoreapp3.0;netcoreapp3.1;net5.0;net6.0;net7.0</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp2.1;netcoreapp3.0;netcoreapp3.1;net5.0;net6.0;net7.0</TargetFrameworks>
+    <TargetFrameworks>netcoreapp2.1;netcoreapp3.0;netcoreapp3.1;net5.0;net6.0;net7.0</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">net461;$(TargetFrameworks)</TargetFrameworks>
+    <RuntimeIdentifiers Condition="'$(OS)' == 'Windows_NT' AND '$(TargetFramework)' == 'net461'">win7-x86;win7-x64</RuntimeIdentifiers>
+  </PropertyGroup>
 
-    <OutputType>Exe</OutputType>
+  <PropertyGroup>
     <Platforms>x64;x86</Platforms>
-    <PlatformTarget>$(Platform)</PlatformTarget>
     <LangVersion>latest</LangVersion>
-
+    <DebugType>full</DebugType>
+    <Optimize>false</Optimize>
```

Comment:
Do you really want to do this? Seems like you should be testing against release-compiled samples no?

EDIT: I see this is what the samples were doing before, but still seems a little strange?

Reply:
For extra context:
1. We were doing that before because reasons (TLDR: the debugger extracts more data if <Optimize>false</Optimize> and we wanted as much data as possible in the `.verified` files. 
2. This is obviously not ideal so in https://github.com/DataDog/dd-trace-dotnet/pull/3504 we have switched to `<Optimize>true</Optimize>`
3. Ultimately we want to run our integration tests both with `Samples.Probes` with optimizations both turned on and turned off, since while we expect 90%+ of production code is RELEASE, there are some customers who run DEBUG builds in production, and we've discovered that the debugger may exhibit different bugs in each case. This PR is actually the first step towards enabling that functionality.